### PR TITLE
[Bugfix][KV Offload] Namespace auto cache dtype by effective dtype

### PR DESCRIPTION
**Pull request by jongukc**
## Purpose

Persistent native KV-offload tiers currently fingerprint the unresolved
`cache_dtype` argument. With the default `cache_dtype="auto"`, FP16 and BF16
engines for the same model therefore share a filesystem or object-store
namespace even though their KV bytes are incompatible. Because both formats
have the same page size, a BF16 engine can accept FP16 pages and silently return
incorrect output after restart.

This change resolves only the `auto` case to `model_config.dtype` before
constructing `OffloadingModelConfig`. Explicit cache formats such as FP8 remain
unchanged because they may share a backing `torch.dtype` while using
incompatible representations.

### Duplicate-work check

I searched open and closed issues and PRs for `FileMapper`, persistent native
offload, `cache_dtype=auto`, FP16/BF16, and cache namespaces. I found no matching
fix. #49261 and #49266 add model-revision identity to the same namespace, but
they do not resolve the literal `auto` dtype. This finding is independent and
persists within one model revision. #48414 fingerprints an opt-in canonical
layout, and #49413 concerns offload-event routing; neither changes cache dtype
identity.

## Test plan

```bash
.venv/bin/python -m pytest -q \
  tests/v1/kv_offload/test_factory.py \
  tests/v1/kv_offload/test_file_mapper.py

.venv/bin/python -m pytest -q tests/v1/kv_offload

.venv/bin/pre-commit run --files \
  vllm/distributed/kv_transfer/kv_connector/v1/offloading/config.py \
  tests/v1/kv_offload/test_factory.py

git diff --check origin/main...HEAD
```

The model-level replay uses `facebook/opt-125m`, V2, FP16/BF16, a fixed
`PYTHONHASHSEED`, and a persistent filesystem tier. It runs cold BF16, populates
the tier with FP16, and then starts BF16 against the same root.

## Test results

Tested on current upstream `main` at `d6dbdb9b0`.

- Focused factory and mapper tests: `40 passed`.
- Complete KV-offload suite: `388 passed`.
- All applicable pre-commit hooks passed, including Ruff and mypy.
- `git diff --check` passed.
- The fixed replay created distinct `float16` and `bfloat16` namespaces.
- The BF16 replay JSON, including token IDs and chosen-token logprobs, was
  byte-identical to the cold BF16 baseline.

Before the fix, the same replay loaded FP16 pages into BF16 and changed the
greedy continuation from `Paris...` to `the first to be a great person...`.

## AI Assistance

OpenAI Codex was used to assist with investigation, implementation, testing,
and PR preparation. The human submitter must review every changed line and be
prepared to explain and defend the change end-to-end.

### CODE REVIEW

**Review by orozery — APPROVED** [review]
Thanks for the fix @jongukc !

**Comment by jongukc** [comment]
@orozery It seems that the failed CI jobs are unrelated to this patch.
(There was a network error during apt command.)
Could you retry them when convenient?

**Comment by orozery** [comment]
> @orozery It seems that the failed CI jobs are unrelated to this patch. (There was a network error during apt command.) Could you retry them when convenient?

I usually just click "update branch" until the flaky CI tests pass.
It seems that currently there is a CI jobs that fails deterministically (`buildkite/ci/v1-sample-plus-logits`).
I think it's best if we watch the current main and wait for it to stabilize, then "update branch" to re-trigger the CI.
